Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/execution/TestCertificateRetrievalScheduler.kt

```diff
@@ -69,21 +68,27 @@ class TestCertificateRetrievalScheduler @Inject constructor(
             .onEach { testsWithoutCert ->
                 Timber.tag(TAG).d("State change: testsWithoutCert=$testsWithoutCert")
                 testsWithoutCert.forEach { test ->
-                    val cert = certificateRepo.requestCertificate(test)
-                    Timber.tag(TAG).v("Certificate was created: %s", cert)
-                    testRepo.markDccAsCreated(test.identifier, created = true)
+                    try {
+                        val cert = certificateRepo.requestCertificate(test)
+                        Timber.tag(TAG).v("Certificate was created: %s", cert)
+                        testRepo.markDccAsCreated(test.identifier, created = true)
+                    } catch (e: Exception) {
+                        Timber.tag(TAG).e(e, "Creation trigger failed.")
+                    }
                 }
             }
-            .catch { Timber.tag(TAG).e(it, "Creation trigger failed.") }
             .launchIn(appScope)
 
         // For each change to the set of existing certificates, check if we need to refresh/load data
         refreshTrigger
             .onEach { checkCerts ->
-                Timber.tag(TAG).d("State change: checkCerts=$checkCerts")
-                if (checkCerts) scheduleWorker()
+                try {
+                    Timber.tag(TAG).d("State change: checkCerts=$checkCerts")
+                    if (checkCerts) scheduleWorker()
```

Comment:
I don't think this could reasonably throw an exception, but catching it doesn't hurt either.

Reply:
i agree, i did it here for consistency with the other occasions, but we could remove it as well.